docs/source/release-notes.md

```diff
@@ -0,0 +1,77 @@
+# Release Notes
+
+NVIDIA Resiliency Extension is a Python package for framework developers and users to implement fault-tolerant features. It improves effective training time by minimizing downtime due to failures and interruptions.
+
+## NVIDIA Resiliency Extension v0.3.0
+
+### Highlights
+
+- Support for Blackwell GPU 
+- ARM based host CPU support 
+- In-process & In-job restart 
+    - Hierarchical in-process and in-job restart support  
+    - Warm spare support 
+    - Detect user application error and avoid restarting 
+- Health checks 
+    - GPU health check based on NVML 
+    - NIC 
+- Checkpointing
+    - Existing capabilities that used to be part of Megatron Core is refactored to be part of NVRx. The checkpointing feature will be maintained as part of NVRx, and Megatron Core and NeMo will use the code from NVRx in the future.
```

Comment:
Please add: 

- Added support for checkpoint metadata caching to improve performance for subsequent checkpoints